### PR TITLE
🏗 performance-urls task 

### PR DESCRIPTION
**Pull request by micajuine-ho**
Created a gulp task to check if localhost urls within `performance/config.json` exist.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -160,6 +161,7 @@ createTask('make-extension', makeExtension);
 createTask('nailgun-start', nailgunStart);
 createTask('nailgun-stop', nailgunStop);
 createTask('performance', performance);
+createTask('performance-urls', performanceUrls);
```
Comment (micajuine-ho) [inline gulpfile.js:164]: Created the task, but not sure how/where I should incorporate this into our CI.

Thoughts @estherkim ?

File: build-system/tasks/performance-urls.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        handler.urls
+          .filter((url) => url.startsWith(LOCAL_HOST_URL))
+          .map((url) =>
+            path.join(__dirname, '../../', url.split(LOCAL_HOST_URL)[1])
+          )
+      );
+      for (const url of urls) {
+        if (!fs.existsSync(url)) {
+          log(colors.red(url + ' does not exist.'));
+          process.exitCode = 1;
```
Comment (micajuine-ho) [inline build-system/tasks/performance-urls.js:52]: Do I need to do anything else here to fail the task?

File: gulpfile.js (hunk 1 above)
Comment (estherkim) [reply gulpfile.js:164]: You can call it here - 
https://github.com/ampproject/amphtml/blob/master/build-system/pr-check/checks.js

File: build-system/tasks/performance-urls.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * @return {!Promise}
+ */
+async function performanceUrls() {
+  return gulp.src([CONFIG_PATH]).pipe(
+    through2.obj(function (file) {
+      let obj;
+      try {
+        obj = JSON.parse(file.contents.toString());
+      } catch (e) {
+        log(colors.yellow(`Could not parse ${CONFIG_PATH}. `));
```
Comment (estherkim) [inline build-system/tasks/performance-urls.js:39]: This should fail the task too, right?

File: build-system/tasks/performance-urls.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        return;
+      }
+      const urls = obj.handlers.flatMap((handler) =>
+        handler.urls
+          .filter((url) => url.startsWith(LOCAL_HOST_URL))
+          .map((url) =>
+            path.join(__dirname, '../../', url.split(LOCAL_HOST_URL)[1])
+          )
+      );
+      for (const url of urls) {
```
Comment (estherkim) [inline build-system/tasks/performance-urls.js:49]: If this succeeds, can we log a `green('SUCCESS')` message?

File: build-system/tasks/performance-urls.js (hunk 2 above)
Comment (estherkim) [reply build-system/tasks/performance-urls.js:52]: Nope, this is fine!

File: build-system/tasks/performance-urls.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  return gulp.src([CONFIG_PATH]).pipe(
+    through2.obj(function (file) {
+      let obj;
+      try {
+        obj = JSON.parse(file.contents.toString());
+      } catch (e) {
+        log(colors.yellow(`Could not parse ${CONFIG_PATH}. `));
+        return;
+      }
+      const urls = obj.handlers.flatMap((handler) =>
```
Comment (estherkim) [inline build-system/tasks/performance-urls.js:42]: nit: rename this to files or filepaths, not urls?

File: gulpfile.js (hunk 1 above)
Comment (micajuine-ho) [reply gulpfile.js:164]: Does adding it here seem appropriate? https://github.com/ampproject/amphtml/blob/master/build-system/pr-check/checks.js#L71

Comment (estherkim) [reply gulpfile.js:164]: Yep, doesn't make sense to create a BUILD_TARGET for this. Make sure to add it to master push builds too, in here https://github.com/ampproject/amphtml/blob/master/build-system/pr-check/checks.js#L47

**Comment by estherkim** [comment]
Adding @rsimha to review :)

File: gulpfile.js (hunk 1 above)
Comment (micajuine-ho) [reply gulpfile.js:164]: Cool, is there a way to test this?

File: build-system/tasks/performance-urls.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/performance-urls.js:53]: Optional: Now that we're on `gulp` v4, here's an alternative way that has the added benefit of printing a clear message without the mostly useless `nodejs` call stack:

https://github.com/ampproject/amphtml/blob/79b463476ec9b7e588bde89c060433bbac54620b/build-system/tasks/check-sourcemaps.js#L37-L46

Note to self: I wonder if it's worth making this a util method that other tasks can also use.

File: gulpfile.js (hunk 1 above)
Comment (rsimha) [reply gulpfile.js:164]: Travis logs show that this succeeded: https://travis-ci.org/github/ampproject/amphtml/jobs/681697643#L428

**Pull request merged**